clientconn_test.go

```diff
@@ -67,6 +68,26 @@ func TestTLSDialTimeout(t *testing.T) {
 	}
 }
 
+func TestDialWithContextCancel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func() {
```

Comment:
what's the point of this goroutine and `done` channel construction? except for a very unlikely race, it always just boils down to calling `cancel`.